akka-docs/src/main/paradox/typed/interaction-patterns.md

```diff
@@ -196,6 +196,40 @@ Java
  * There can only be a single response to one `ask` (see @ref:[per session child Actor](#per-session-child-actor))
  * When `ask` times out, the receiving actor does not know and may still process it to completion, or even start processing it after the fact
 
+## Send Future result to self
+
+When using an API that returns a @scala[`Future`]@java[`CompletionStage`] from an actor it's common that you would
+like to use the value of the in the actor when the @scala[`Future`]@java[`CompletionStage`] is completed. For
+this purpose the `ActorContext` provides a `pipeToSelf` method.
+
+**Example:**
+
+![pipe-to-self.png](./images/pipe-to-self.png)
+
+An actor, `CustomerRepository`, is invoking a method on `CustomerDataAccess` that returns a @scala[`Future`]@java[`CompletionStage`].
```

Comment:
Looks good. My only thought is that it is a big chunk of code that could be intimidating for showing one API.

Reply:
The other interaction pattern examples are equally large.
I see it more as a way to make the patterns more complete by also including this pretty common interaction type with an external Future. Purpose is not so much as reference docs of the `pipeToSelf`.